src/nodes.coffee

```diff
@@ -3238,12 +3267,10 @@ exports.Op = class Op extends Base
   constructor: (op, first, second, flip, {@invertOperator, @originalOperator = op} = {}) ->
     super()
 
-    if op is 'do'
-      return Op::generateDo first
     if op is 'new'
-      if (firstCall = first.unwrap()) instanceof Call and not firstCall.do and not firstCall.isNew
-        return firstCall.newInstance()
-      first = new Parens firstCall if firstCall instanceof Code and firstCall.bound or firstCall.do
+      if ((firstCall = unwrapped = first.unwrap()) instanceof Call or (firstCall = unwrapped.base) instanceof Call) and not firstCall.do and not firstCall.isNew
```

Comment:
> It feels like the thing that would clean this up would be to always transform `new` from an `Op` to a `Call` (which is what it really is) up front

Yes. I feel like we could do a better job with this in the grammar, so that there’s less hackiness here. For example why are we updating the location data to expand to the left to pull in the `new `? We don’t need such shenanigans for any other node. The grammar should handle that kind of thing I would think. (Not saying you did anything wrong, this is very old code, you’re just making it obvious why it’s bad 😄)